### DISCUSSION TITLE
anyone know why firefox is not installing?

### DISCUSSION THREAD

**Discussion by kaisers0s3** [chocolatey/home#320] [Q&A]
Just a warning- i am just getting into the coding world so all i have is google lol
here is my code-
function install-Browsers(){
  msg "Download/Install Firefox and Google Chrome" 

  # Install packages
  $chocoPackages = "Firefox", "GoogleChrome","microsoft-edge"
    
  foreach ($package in $chocoPackages)
    {  
    Write-Host "Installing Package: $package" -ForegroundColor Cyan
    choco install -y $package --no-progress
  }

  #Add cert to the Chrome.
  msg "Installing SSL certs in browsers"
  
}

here is my jenkins console output-

==========> Download/Install Firefox and Google Chrome <==========

Installing Package: Firefox
Chocolatey v2.2.2
Installing the following packages:
Firefox
By installing, you accept licenses for the packages.

Firefox v126.0.0 [Approved]
Firefox package files install completed. Performing other installation steps.
Using locale 'en-US'...
Downloading Firefox 64 bit
  from 'https://download.mozilla.org/?product=firefox-126.0-ssl&os=win64&lang=en-US'

Download of Firefox Setup 126.0.exe (61.42 MB) completed.
Hashes match.
Installing Firefox...
Chocolatey timed out waiting for the command to finish. The timeout
 specified (or the default value) was '2700' seconds. Perhaps try a
 higher `--execution-timeout`? See `choco -h` for details.
  Firefox may be able to be automatically uninstalled.
The install of Firefox was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\Firefox\tools\chocolateyInstall.ps1'.
 See log for details

**Comment by pauby** [comment]
This is a PowerShell issue and now a Chocolatey CLI one.

The error is:

```
Chocolatey timed out waiting for the command to finish. The timeout
specified (or the default value) was '2700' seconds. Perhaps try a
higher --execution-timeout? See choco -h for details.
```

Chocolatey CLI is timing out.

Have your run the command `choco install -y Firefox --no-progress` outside of the script? Have you looked at the Chocolatey CLI logs?
